packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -168,7 +190,26 @@ export class AssociationManager {
     return `- Portfolio: ${portfolio.node.path} | Product: ${product.node.path}`;
   }
 
+  private static formatAssertions(scope: cdk.Resource, assertions : Assertion[]): {[key: string]: string}[] {
```

Comment:
`stack` is a property of `IResource`. Change the signature to have `scope` of that type, and you'll avoid all of this casting.